rust/gen-cpp-doc/src/main.rs

```diff
@@ -761,8 +763,8 @@ fn parse<'tu>(
 
 fn modify_book(book: &mut mdbook::book::Book, mut items: Vec<Item>) {
     let cpp_doc_re = Regex::new(r"[{][{] *#cpp-doc +(::[^ ]+)+ *[}][}]").unwrap();
-    // I got lazy here. ` within the diagram will cause svg_bob_re to not match.
-    let svg_bob_re = Regex::new(r"```svgbob([^`]+)```").unwrap();
+    let svg_bob_re = Regex::new(r"```svgbob((.|\n)*?)```").unwrap();
```

Comment:
`[^]` is a more standard way to match any character. The exact set of characters excluded from `.` isn't completely consistent across regex implementations.

Reply:
If I use `[^]` then Rust regex complains:
```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    ```svgbob([^]*?)```
              ^^^
error: unclosed character class
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
)
```

The [regex](https://docs.rs/regex/latest/regex/#matching-one-character) crate docs specify that `.` matches "any character except new line (includes new line with s flag)".

Reply:
A bit of poking around shows that there are 3 flags that affect the behavior of "."
- 'R': excludes \r\n instead of \n
- 'U': Matches utf8 code points instead of bytes (on by default)
- 's': matches anything including newlines
Based on this, I suggest setting the 's' flag.